nmostesting/GenericTest.py

```diff
@@ -503,6 +521,93 @@ def do_test_authorization(self, api_name, test_name, error_code=401, error_type=
             return test.DISABLED("This test is only performed when an API supports Authorization and 'ENABLE_AUTH' "
                                  "is True")
 
+    def do_test_no_matching_public_key_authorization(self, api_name):
+        api = self.apis[api_name]
+        url = "{}".format(api["url"].rstrip("/"))
+        test = Test("GET /x-nmos/{}/{} (No Matching Public Keys)".format(api_name,
+                                                                         api["version"]), self.auto_test_name(api_name))
+
+        if self.authorization:
+            # start the mock secondary Authorization server
+            secondary_authorization_server = AuthServer(self.secondary_auth)
+            secondary_authorization_server.start()
+
+            # generate token
+            token = self.secondary_auth.generate_token([api_name])
+
+            fail = None
+            warning = None
+            headers = {"Authorization": "Bearer {}".format(token)}
+            valid, response = self.do_request("GET", url, headers=headers)
+
+            if not valid:
+                fail = response
+            elif response.status_code != 401 and response.status_code != 503:
+                fail = "Incorrect response code, expected 401 or 503. Received {}".format(response.status_code)
+                if "WWW-Authenticate" not in response.headers:
+                    fail = "Authorization error responses must include a 'WWW-Authenticate' header"
+                elif not response.headers["WWW-Authenticate"].startswith("Bearer "):
+                    fail = "'WWW-Authenticate' response header must begin 'Bearer'"
+                else:
+                    error_code = response.status_code
+                    valid, message = self.check_error_response("GET", response, error_code)
+                    if not valid:
+                        fail = message
+
+            # if node responds with 401, node should attempt to obtain the missing public key via the the token
+            # iss claim, as specified in RFC 8414 section 3.
```

Comment:
As Michael mentioned in AMWA-TV/is-10#111, I don't see anything about fetching the missing public key in RFC 8414, section 3 or otherwise. Do you have the correct reference so we can fix IS-10 too?

In fact, I've also seen guidance that the `iss` claim should not be blindly followed but should be subject to an allow list... e.g. see https://curity.io/resources/learn/jwt-best-practices/#5-always-check-the-issuer.

Reply:
see https://github.com/AMWA-TV/is-10/issues/111#issuecomment-1433698690
 
Regarding `iss` claim, in a real system it is always an URL using `HTTPS` scheme, and the token issuer (the Authorization server) always has valid certificate to identify it from being a dodgy server.

Reply:
It seems to me that ensures that the server is who it says it is, it doesn't mean that you should trust that server as an issuer of JWT for this system?

Reply:
So, the slightly confusing reference to RFC 8414 Section 3 is a spec issue covered by https://github.com/AMWA-TV/is-10/issues/111, and the trust issue is covered by https://github.com/AMWA-TV/is-10/issues/114.